core/templates/domain/story/editable-story-backend-api.service.ts

```diff
@@ -16,32 +16,45 @@
  * @fileoverview Service to send changes to a story to the backend.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-require('domain/story/story-domain.constants.ajs.ts');
-
-angular.module('oppia').factory('EditableStoryBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService',
-  'EDITABLE_STORY_DATA_URL_TEMPLATE', 'STORY_PUBLISH_URL_TEMPLATE',
-  'STORY_URL_FRAGMENT_HANDLER_URL_TEMPLATE',
-  'VALIDATE_EXPLORATIONS_URL_TEMPLATE',
-  function(
-      $http, $q, UrlInterpolationService,
-      EDITABLE_STORY_DATA_URL_TEMPLATE, STORY_PUBLISH_URL_TEMPLATE,
-      STORY_URL_FRAGMENT_HANDLER_URL_TEMPLATE,
-      VALIDATE_EXPLORATIONS_URL_TEMPLATE) {
-    var _fetchStory = function(storyId, successCallback, errorCallback) {
-      var storyDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_STORY_DATA_URL_TEMPLATE, {
-          story_id: storyId
-        });
-
-      $http.get(storyDataUrl).then(function(response) {
-        var story = angular.copy(response.data.story);
-        var topicName = angular.copy(response.data.topic_name);
-        var storyIsPublished = response.data.story_is_published;
-        var skillSummaries = angular.copy(response.data.skill_summaries);
-        var topicUrlFragment = response.data.topic_url_fragment;
-        var classroomUrlFragment = response.data.classroom_url_fragment;
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { StoryDomainConstants } from 'domain/story/story-domain.constants';
+
+interface FetchStoryResponse {
+  'story':object;
+  'topic_name':string;
+  'story_is_published':boolean;
+  'skill_summaries':object;
```

Comment:
Mention the exact type instead of object.

Reply:
@nishantwrp done